src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -2045,6 +2054,9 @@ If such a class is used as a base class and if the deriving class defines a dest
   <data name="ERR_PartialMethodMustHaveLatent" xml:space="preserve">
     <value>No defining declaration found for implementing declaration of partial method '{0}'</value>
   </data>
+  <data name="ERR_PartialMethodTupleNameDifference" xml:space="preserve">
+    <value>Both partial method declarations, '{0}' and '{1}', must use the same tuple element names.</value>
```

Comment:
I changed the error code, but I'm not sure how to best phrase the error message.

The one I introduced spells out the two method declarations that clash, which I think is nice. But the phrasing is clunky :-(
```
   // (5,18): error CS8221: Both partial method declarations, 'C.M2((int a, int b))' and 'C.M2((int, int))', must use the same tuple element names.
                //     partial void M2((int a, int b) x);
```
```
  <data name="ERR_PartialMethodInconsistentTupleNames" xml:space="preserve">
    <value>Both partial method declarations, '{0}' and '{1}', must use the same tuple element names.</value>
  </data>
  <data name="ERR_PartialMethodInconsistentConstraints" xml:space="preserve">
    <value>Partial method declarations of '{0}' have inconsistent type parameter constraints</value>
  </data
```